tensorboard/webapp/widgets/line_chart_v2/line_chart_internal_utils.ts

```diff
@@ -22,44 +23,55 @@ import {isWebGl2Supported} from './lib/utils';
 /**
  * Returns extent, min and max values of each dimensions, of all data series points.
  *
+ * When ignoreYOutliers is true, it will filter out values that are approximately [1] less
+ * than 5th and greater than 95th percentile when calculating y-extent.
+ *
  * Note that it excludes auxillary data points and invisible data series.
  *
- * TODO(stephanwlee): add support for ignoreOutlier.
+ * [1]: Uses R-7 method for approximation: https://github.com/d3/d3-array#quantile
  */
 export function computeDataSeriesExtent(
   data: DataSeries[],
-  metadataMap: DataSeriesMetadataMap
+  metadataMap: DataSeriesMetadataMap,
+  ignoreYOutliers: boolean
 ): {x: [number, number] | undefined; y: [number, number] | undefined} {
-  let xMin = Infinity;
-  let xMax = -Infinity;
-  let yMin = Infinity;
-  let yMax = -Infinity;
-
-  let xExtentChanged = false;
-  let yExtentChanged = false;
+  let yPoints: number[] = [];
+  const xPoints: number[] = [];
 
+  let pointIndex = 0;
   for (const {id, points} of data) {
     const meta = metadataMap[id];
     if (!meta || meta.aux || !meta.visible) continue;
 
     for (let index = 0; index < points.length; index++) {
       const {x, y} = points[index];
-      if (!Number.isNaN(x)) {
-        xMin = Math.min(xMin, x);
-        xMax = Math.max(xMax, x);
-        xExtentChanged = true;
+      if (Number.isFinite(x)) {
+        xPoints.push(x);
```

Comment:
Optional but since we still only need the min and max, it seems more efficient to stick with manually tracking those, rather than using an extra O(n) memory to create xPoints and then another O(n) compute to traverse it in `d3.extent` below?  I guess in practice maybe it makes little difference but if we're trying to emphasize perf here it seems a little weird to make this slower.

perhaps slightly unnecessary to accumulate a copy of the array of points?  I guess in practice this is probably pretty cheap but if we're trying to emphasize performance it